src/jit/gentree.cpp

```diff
@@ -8244,6 +8575,14 @@ GenTreePtr GenTree::GetChild(unsigned childNum)
     {
         return AsUnOp()->gtOp1;
     }
+    // Special case for assignment of dynamic block.
+    // This code is here to duplicate the former case where the size may be evaluated prior to the
+    // source and destination addresses. In order to do this, we treat the size as a child of the
+    // assignment.
+    if ((OperGet() == GT_ASG) && (gtOp.gtOp1->OperGet() == GT_DYN_BLK) && (childNum == 2))
+    {
+        return gtOp.gtOp1->AsDynBlk()->gtDynamicSize;
+    }
     else if (OperIsBinary())
```

Comment:
It's unfortunate that some of these dynblk special cases introduce checks that are probably more expensive than the cheaper and more common cases that follow. Same is true in the GTK_SMPOP check I saw elsewhere that is now preceded by a relatively more expensive condition. I think GetChild is relatively hot, too.